tools/wt_to_mdb_bson.py

```diff
@@ -165,42 +165,55 @@ def execute_wt(mode, wtpath, uri):
             [wtpath, "-r", "-C", "log=(compressor=snappy,path=journal/)", "printlog", "-u", "-x"], universal_newlines=True).splitlines()
 
 def main():
-    if len(sys.argv) < 3:
-        util_usage()
-        exit()
-
-    if sys.argv[1] != '-m':
-        print('A mode must be specified with -m.')
-        util_usage()
-
-    mode_str = sys.argv[2]
+    parser = argparse.ArgumentParser(
+        prog='wt_to_mdb_bson',
+        description="A tool to convert WT contents to MognoDB's BSON representation"
+    )
+    parser.add_argument("-m", "--mode",
+       required=True,
+       help="The intended mode that the wt util operated in or will be executed using",
+       choices=["dump", "verify", "printlog"]
+    )
+    parser.add_argument("-j", "--json",
+        action="store_true",
+        help="Outputs the BSON as proper JSON using the canonical JSON format"
+    )
+    parser.add_argument("-f", "--wt-path",
+        help="The location of the wt util"
+    )
+    parser.add_argument("uri",
+       nargs='?',
+       default=None
+    )
+
+    args = parser.parse_args()
+
+    mode_str = args.mode
     if mode_str == 'dump':
         mode = Mode.DUMP
     elif mode_str == 'verify':
         mode = Mode.VERIFY
-    elif mode_str == 'printlog':
-        mode = Mode.PRINTLOG
     else:
```

Comment:
I would still like to see some kind of message printed out when an invalid mode is specified. With your current changes, it looks like anything that is not `dump` or `verify` will be interpreted as the `printlog` mode.

Reply:
I refactored the script to use argparse which does take care of this issue since the only supported values are the three values expected. If anything else is provided the program will exit with an error